lib/active_merchant/billing/gateways/quickpay.rb

```diff
@@ -1,371 +1,25 @@
-        add_autocapture(post, false)
-        add_fraud_parameters(post, options) if action.eql?(:authorize)
-        add_testmode(post)
-
-        commit(action, post)
-      end
-
-      def purchase(money, credit_card_or_reference, options = {})
-        post = {}
-
-        action = recurring_or_authorize(credit_card_or_reference)
-
-        add_amount(post, money, options)
-        add_creditcard_or_reference(post, credit_card_or_reference, options)
-        add_invoice(post, options)
-        add_fraud_parameters(post, options) if action.eql?(:authorize)
-        add_autocapture(post, true)
-
-        commit(action, post)
-      end
-
-      def capture(money, authorization, options = {})
-        post = {}
-
-        add_finalize(post, options)
-        add_reference(post, authorization)
-        add_amount_without_currency(post, money)
-        commit(:capture, post)
-      end
-
-      def void(identification, options = {})
-        post = {}
-
-        add_reference(post, identification)
+      self.abstract_class = true
 
-        commit(:cancel, post)
-      end
-
-      def refund(money, identification, options = {})
-        post = {}
-
-        add_amount_without_currency(post, money)
-        add_reference(post, identification)
-
-        commit(:refund, post)
-      end
-
-      def credit(money, identification, options = {})
-        ActiveMerchant.deprecated CREDIT_DEPRECATION_MESSAGE
-        refund(money, identification, options)
-      end
-
-      def store(creditcard, options = {})
-        post = {}
-
-        add_creditcard(post, creditcard, options)
-        add_amount(post, 0, options) if @protocol >= 7
-        add_invoice(post, options)
-        add_description(post, options)
-        add_fraud_parameters(post, options)
-        add_testmode(post)
-
-        commit(:subscribe, post)
-      end
-
-      private
-
-      def add_amount(post, money, options = {})
-        post[:amount]   = amount(money)
-        post[:currency] = options[:currency] || currency(money)
-      end
-
-      def add_amount_without_currency(post, money, options = {})
-        post[:amount] = amount(money)
-      end
-
-      def add_invoice(post, options)
-        post[:ordernumber] = format_order_number(options[:order_id])
-      end
-
-      def add_creditcard(post, credit_card, options)
-        post[:cardnumber]     = credit_card.number
-        post[:cvd]            = credit_card.verification_value
-        post[:expirationdate] = expdate(credit_card)
-        post[:cardtypelock]   = options[:cardtypelock] unless options[:cardtypelock].blank?
-        post[:acquirers]      = options[:acquirers] unless options[:acquirers].blank?
-      end
-
-      def add_reference(post, identification)
-        post[:transaction] = identification
-      end
+      def self.new options = {}
+        options.fetch(:login) rescue raise ArgumentError.new("Missing required parameter: login")
```

Comment:
@ntalbott This custom check is implemented because `requires!` is an instance method. If you still feel we should change this, do you have any advice as to how we could best do this?